### ISSUE TITLE
writeToDisk not working with webpack-serve (multiple configuration setup)

### ISSUE THREAD

**Issue by DBosley** [webpack/webpack-dev-middleware#315]
Operating System: Ubuntu 18.04
Node Version: 8.11.1
NPM Version: 6.2.0
webpack Version: 4.12.0
webpack-hot-client Version: 4.0.3

- [ ] This is a **feature** request
- [x] This is a **bug**

### Code

For security and NDA reasons I can't share my full webpack config, but below are the specific parts that I think are relevent. If you need more, I'll try to provide what you need.
Paired with webpack-serve configs:
```js
serve:{ 
  host: '0.0.0.0',
  port: 3000,
  hot:
   { host: { client: 'website.localhost?', server: '0.0.0.0' },
     https: true,
     port: 3001,
     allEntries: true },
  content: '/static',
  add: (app, middleware, options) => {
      const router = new Router();
      router.get(
        '/static/*',
        convert(
          proxy({
            target: 'http://localhost:3000/',
            pathRewrite: {
              '^/static': ''
            }
          })
        )
      );
      middleware.webpack();
      middleware.content();

      app.use(logger());
      app.use(router.routes());
    },
  devMiddleware: { writeToDisk: true } }
```

### Expected Behavior
`devMiddleware: {writeToDisk: true }}`
Should write bundled code to the disk.

### Actual Behavior
Does not write to disk.

### For Bugs; How can we reproduce the behavior?
As far as I can tell, all you need to do is add `devMiddleware: {writeToDisk: true}` to the webpack `serve` config object in the webpack config.

**Comment by DBosley** [comment]
Thanks for the info about the middleware.webpack call. I changed it to this:

```js
      middleware.webpack().then(()=>{
        middleware.content();
        app.use(logger());
        app.use(router.routes());
      });
```
But that didn't fix my problem with writeToDisk.

Next, I tried using this minimal configuration:
```js
serve:{
  devMiddleware:{
    writeToDisk: true
  }
}
```
and it still doesn't output the bundled files.
When running webpack directly, the files are output correctly.

**Comment by DBosley** [comment]
Update on this issue:
I realized the reason it wasn't working at all was because my webpack-serve was out of date. I've updated that and it mostly works. There does still seem to be an issue when using a multiple-config webpack build. I only want one of the configurations I'm exporting to `writeToDisk` but when I declare it on the second config in the array it doesn't work. If I put the option in the first config it works fine but every single bundle is written to disk.
